test/widgets/inbox_test.dart

```diff
@@ -310,6 +345,30 @@ void main() {
           checkAppearsUncollapsed(tester, findSectionContent);
         });
 
+        testWidgets('collapse all-DMs section after scroll', (tester) async {
+          await setupVarious(tester);
+
+          final listFinder = find.byType(Scrollable);
+          final findSectionContent =
+              find.text(eg.otherUser.fullName).hitTestable();
+
+          // Scroll the [StickyHeaderListView] enough so that
+          // the [_AllDmsSection] shows a sticky header
+          await dragUntilInvisible(
+              tester, findSectionContent, listFinder, const Offset(0, -50));
```

Comment:
This code on its own doesn't ensure that there's a sticky header from that section — it could be that when this completes, that section is out of view entirely.

This logic does work (because of the check that comes next), but I'd tell the story a bit differently:
```dart
          // Scroll part of [_AllDmsSection] offscreen…
          await dragUntilInvisible(
              tester, findSectionContent, listFinder, const Offset(0, -50));
          // … and check its header is still present (which must therefore
          // be as a sticky header).
          check(findAllDmsHeaderRow(tester)).isNotNull();
```

Reply:
In fact better yet, making it a bit more self-contained:
```dart
          // Scroll part of [_AllDmsSection] offscreen…
          await dragUntilInvisible(tester,
            find.text(eg.otherUser.fullName).hitTestable(),
            find.byType(Scrollable),
            const Offset(0, -50),
          );
          // … and check its header is still present (which must therefore
          // be as a sticky header).
          check(findAllDmsHeaderRow(tester)).isNotNull();
```